### DISCUSSION TITLE
JWT auth - session undefined

### DISCUSSION THREAD

**Discussion by louis-xborg** [supabase/supabase#9787] [Questions]
Hi there, 

I am trying to authenticate the user with a JWT. while it used to work with setAuth in v1, I can't seem to authenticate the user with the global authorization header in v2. 

I pass my JWT in the headers, yet, no session gets instantiated. getSession return null. 

When I call client.auth.getUser(token), it returns 500 internal server error. error_id: "aaa46750-3dcb-4f7d-8871-283d4e452dbf"

I assume there is an issue with how I set my session, I currently use Authentication headers: 
createClient(Environment.Supabase.URL, Environment.Supabase.ANON_KEY, {
  ...clientOptions,
  global: {
    headers: {
      Authorization: `Bearer ${accessToken}`,
    },
  },
});

Where is the latest documentation on setting up sessions with custom JWTs?

**Comment by silentworks** [comment]
Can you explain what you mean by setting up session with custom JWTs please? When you create the new client you need to use that instance from there on and not the existing one.

```js
const newClient = createClient(Environment.Supabase.URL, Environment.Supabase.ANON_KEY, {
  ...clientOptions,
  global: {
    headers: {
      Authorization: Bearer ${accessToken},
    },
  },
});
```
and now you can do
```js
const { session } = await newClient.auth.getSession()
```

**Reply by louis-xborg** [reply]
Thanks a lot for your reply. 

However, the issue is that when I create a new client, (with createClient and access token), no session gets created. When I call await newClient.auth.getSession(), it returns null. 

I have a hard time locating where the issue could come from

**Reply by louis-xborg** [reply]
If that may help, here is the JWT that I use to log in to the client: 
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJhdXRoZW50aWNhdGVkIiwiZXhwIjoxNjY2NjIxNTc4LCJzdWIiOiI4NzI5ZDRlOS0wZmVjLTQ1MDEtYjk0ZC03NzQyMTE5YWUyZDAiLCJ1c2VyX21ldGFkYXRhIjp7ImlkIjoiODcyOWQ0ZTktMGZlYy00NTAxLWI5NGQtNzc0MjExOWFlMmQwIiwid2FsbGV0IjoiMHhhOEFEM2I3NDA3MDMxMDkxRjI3YWYzQkZlREYxNzBlREM3N0NmNzYxIn0sInJvbGUiOiJhdXRoZW50aWNhdGVkIiwiaWF0IjoxNjY2NjE3OTc4fQ.PPA4mBcbuJ4dCq6t3Y2TXoIDWy_oIwld4gxWkkWHuKI

**Reply by Gitnock** [reply]
Hi @xborgofficial ,  I have been trying to solve this exact problem for a while now did you manage to find a solution.

**Reply by ioucyf** [reply]
I'm having the same problem.
